File: core/dotnet/src/serialization/ParseNodeFactoryRegistry.cs
using System;
using System.Collections.Generic;
using System.IO;
using Kiota.Abstractions;
using Kiota.Abstractions.Serialization;

namespace KiotaCore.Serialization {
    public class ParseNodeFactoryRegistry : IParseNodeFactory {
        public Dictionary<string, IParseNodeFactory> ContentTypeAssociatedFactories {get; set;} = new Dictionary<string, IParseNodeFactory>();
        public IParseNode GetRootParseNode(string contentType, Stream content) {
            if(string.IsNullOrEmpty(contentType))
                throw new ArgumentNullException(nameof(contentType));
            _ = content ?? throw new ArgumentNullException(nameof(content));

            if(ContentTypeAssociatedFactories.ContainsKey(contentType))
                return ContentTypeAssociatedFactories[contentType].GetRootParseNode(contentType, content);
            else
                throw new InvalidOperationException($"Content type {contentType} does not have a factory registered to be parsed");
        }
    }
}

File: core/typescript/src/serialization/parseNodeFactoryRegistry.ts
import { ParseNode, ParseNodeFactory } from "@microsoft/kiota-abstractions";

export class ParseNodeFactoryRegistry implements ParseNodeFactory {
    public contentTypeAssociatedFactories = new Map<string, ParseNodeFactory>();
    public getRootParseNode(contentType: string, content: ArrayBuffer): ParseNode {
        if(!contentType) {
            throw new Error("content type cannot be undefined or empty");
        }
        if(!content) {
            throw new Error("content cannot be undefined or empty");
        }
        const factory = this.contentTypeAssociatedFactories.get(contentType);
        if(factory) {
            return factory.getRootParseNode(contentType, content);
        } else {
            throw new Error(`Content type ${contentType} does not have a factory registered to be parsed`);
        }
    }

}